subprojects/configuration-cache/src/main/kotlin/org/gradle/configurationcache/initialization/ConfigurationCacheProblemsListener.kt

```diff
@@ -56,7 +61,32 @@ class DefaultConfigurationCacheProblemsListener internal constructor(
     private val workExecutionTracker: WorkExecutionTracker,
     private val featureFlags: FeatureFlags,
     private val inputTrackingState: InputTrackingState,
-) : ConfigurationCacheProblemsListener {
+    instrumentedExecutionAccessListener: InstrumentedExecutionAccessListener,
+) : ConfigurationCacheProblemsListener, Stoppable {
+
+    init {
```

Comment:
🔧 I'd move the `init` and `close` to a separate listener service with the sole responsibility of installing and removing the execution access listener.